ocaml/xapi/repository.ml

```diff
@@ -728,13 +788,17 @@ let apply_updates ~__context ~host ~hash =
   assert_no_host_pending_mandatory_guidance ~__context ~host ;
   try
     let repository = get_single_enabled_update_repository ~__context in
+    let repository_name = get_repository_name ~__context ~self:repository in
     if hash = "" || hash <> Db.Repository.get_hash ~__context ~self:repository
     then
       raise Api_errors.(Server_error (updateinfo_hash_mismatch, [])) ;
     with_pool_repositories (fun () ->
         let updates_info =
           parse_updateinfo ~__context ~self:repository ~check:true
         in
+        if Helpers.is_pool_master ~__context ~host then
+          maybe_set_restart_for_all_vms ~__context ~repository_name
```

Comment:
Shouldn't we do this _after_ applying the updates?

Reply:
I think this is an action on behalf of the whole pool. The coordinator should not be a special one. And doing it before applying any updates would make sure it will be done anyway without being impacted by possible failures in applying updates.
May I please know the specific concerns on this?

Reply:
Well I thought that if the updates fails to apply already on the coordinator, then restarting VMs becomes unnecessary.

Reply:
OK. This makes sense. Once the coordinator has been updated, the pool update process has started actually.

Reply:
Sorry, what is the difference as the RestartVM is set after applying updates? I think it is the same as the result of applying update is not considered when setting RestartVM guidance?

Reply:
Thanks for the explanation, RestartVM guidance would not be added in case there is a failure in applying updates as this process would be quit with an exception.